src/klein/_isession.py

```diff
@@ -328,15 +320,33 @@ def finalize():
         """
 
 
+class _IRequestLifecycle(Interface):
+    """
+    Interface for adding hooks to the phases of a request's lifecycle.
+    """
+
+
+if TYPE_CHECKING:
+    from typing import Union
+    from ._requirer import RequestLifecycle
+
+    IRequestLifecycle = Union[_IRequestLifecycle, RequestLifecycle]
```

Comment:
In a future PR, I guess you can explore using the mypy zope plugin for klein.
I introduce the use of the mypy zope plugin in Twisted core, so you can get some ideas from that.

Reply:
Definitely.  When I last checked, it wasn't ready, and I wanted to get this big-diff mechanical crap done first.

I did notice you are using it in Twisted, so I suspect it'll work here now.

Reply:
#304